test/sanitytest.c

```diff
@@ -76,6 +77,38 @@ static int test_sanity_unsigned_conversion(void)
 
 static int test_sanity_range(void)
 {
+    /* Verify some types are the correct size */
+    if (!TEST_size_t_eq(sizeof(int8_t), 1)
+            || !TEST_size_t_eq(sizeof(uint8_t), 1)
+            || !TEST_size_t_eq(sizeof(int16_t), 2)
+            || !TEST_size_t_eq(sizeof(uint16_t), 2)
+            || !TEST_size_t_eq(sizeof(int32_t), 4)
+            || !TEST_size_t_eq(sizeof(uint32_t), 4)
+            || !TEST_size_t_eq(sizeof(int64_t), 8)
+            || !TEST_size_t_eq(sizeof(uint64_t), 8)
+#ifdef UINT128_MAX
+            || !TEST_size_t_eq(sizeof(int128_t), 16)
+            || !TEST_size_t_eq(sizeof(uint128_t), 16)
+#endif
+            || !TEST_size_t_eq(sizeof(char), 1)
+            || !TEST_size_t_eq(sizeof(unsigned char), 1))
+        return 0;
+
+    /* We want our long longs to be at least 64 bits */
+    if (!TEST_size_t_ge(sizeof(long long int), 8)
+            || !TEST_size_t_ge(sizeof(unsigned long long int), 8))
```

Comment:
Hmmm. I don't think "long long" is a C89/C90 type. We have a number of instances of this in the code but virtually all of them are in architecture specific code where we *know* the type exists. There are a handful of other instances, but I think they are all in *new* code. I wonder if we are setting up a problem for ourselves by introducing this type? Will it be supported everywhere?

Reply:
int32_t, et al aren't part of C89 either.  In theory long long was permitted in C89, although nobody implemented it and it wasn't standardised until later.

Reply:
> int32_t, et al aren't part of C89 either.

True, but we made a conscious decision to include them and have platform specific typedefs for platforms that don't have them. "long long" seems to have slipped in by accident. Do we really need it?

> In theory long long was permitted in C89,

I'm not sure what "permitted" means. My copy of the spec says "There are four signed integer types, designated as signed char, short int, int and long int"....similarly there are the unsigned int variants. I don't see anything there explicitly permitting "long long".

Reply:
_long_ was a type modifier not a type and in theory could be specified multiple times.  That no compiler back then supported it makes it moot.

We should be able to remove long long references in favour of int64_t.  I'm not convinced we should do this for 3.0 however.

Reply:
BTW poly1305.c seems to use long long as a fall back case: not Windows, not ARM64, sized ints not defined.

Reply:
> long was a type modifier not a type and in theory could be specified multiple times.

That's not correct. Section 6.5.2 of the spec lists the allowed variants. For long it lists:

" - long, signed long, long int, signed long int
  - unsigned long, or unsigned long int"
 
> We should be able to remove long long references in favour of int64_t. I'm not convinced we should do this for 3.0 however.

I only see 4 instances in non-architecture specific code that look dubious to me, so it might not be that hard to do if we choose to.

Reply:
long double was a thing back in the day too.

Reply:
> long double was a thing back in the day too.

Yes, that exists in the spec too, but I only listed the integral types above.